### PR TITLE
Refactor - Clean Code JavaScript - Variables

### PR DESCRIPTION
**Pull request by mirhamasala**
## 📝 Description

This update includes a series of refactors aimed at improving code readability, maintainability, and consistency according to clean code principles - [Chapter 2 - Variables](https://github.com/ryanmcdermott/clean-code-javascript?tab=readme-ov-file#variables). The changes focus on consistent naming conventions, type safety enhancements, and the organization of constants and configurations.

- **Type:** Refactor

## 🛠️ Key Changes

- **Refactor 1:** Standardized variable naming conventions to improve clarity and maintainability.
- **Refactor 2:** Enhanced type safety by introducing TypeScript types for consistent and error-free usage.
- **Refactor 3:** Centralized external URLs and other constants for easier management and reduced redundancy.

## 🔖 Resources

- [Clean Code JavaScript](https://github.com/ryanmcdermott/clean-code-javascript)

### CODE REVIEW

File: src/app/_data/homepage/filecoinEcosystemData.ts [hunk 1]
```diff
@@ -1,15 +1,33 @@
...
+  heading: {
+    title: string
+    icon: IconProps['component']
+  }
+  description: string
+  cta: {
+    href: string
+    text: string
+  }
+}
```
Comment (barbaraperic) [inline src/app/_data/homepage/filecoinEcosystemData.ts:18]: are we starting to use interface?

File: src/app/_components/Icon.tsx [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { type Icon } from '@phosphor-icons/react'
+import type { Icon } from '@phosphor-icons/react'
```
Comment (barbaraperic) [inline src/app/_components/Icon.tsx:1]: when do we put `type` outside of the bracket, and when inside? They are both allowed, just curious if we should follow any pattern?

File: src/app/_data/homepage/filecoinEcosystemData.ts (hunk 1 above)
Comment (barbaraperic) [inline src/app/_data/homepage/filecoinEcosystemData.ts:18]: in what cases do we prefer interface over type? Sorry if I missed a discussion on this

**Review by barbaraperic — APPROVED** [review]

File: src/app/_components/Icon.tsx (hunk 2 above)
Comment (mirhamasala) [reply src/app/_components/Icon.tsx:1]: We cam use `import type { TypeName }` when importing only types to clearly indicate no runtime code is being imported. For mixed imports, we can use `import { type TypeName, AnotherItem }` to keep related imports together and maintain organized code.

File: src/app/_data/homepage/filecoinEcosystemData.ts (hunk 1 above)
Comment (mirhamasala) [reply src/app/_data/homepage/filecoinEcosystemData.ts:18]: Prefer `interface` when you need to define an object type that might be extended or merged, as interfaces support declaration merging and are more flexible for hierarchical design. Use `type` for defining unions, intersections, or more complex types, as it’s more versatile for combining types and creating aliases.

**Pull request merged**